### DISCUSSION TITLE
Construct a crs with local calibration?

### DISCUSSION THREAD

**Discussion by originlake** [pyproj4/pyproj#1283] [Q&A]
A lot of GPS device allows to have local calibration of the coordinate system, which mostly would add an extra step of an affine transformation to the final coordinates. I'm building software to accept the wkt/proj4 string to pass around, it can take GPS device's export and parse it as a wkt or proj4 string and send it to the backend and any other place that needs to do some coordinate conversion. While most of the parts in the GPS exported file follow the standard, I'm not sure how to deal with the local calibration. 
Ideally, I want it to be constructed as a single crs and work with pyproj CRS system directly. But I couldn't find a way to achieve it. I'm currently using a workaround by creating the whole pipeline and adding one step `+proj=affine` at the end, which works, but it's not very convenient to transform or inverse transform to any other normal CRS. 

Does anybody have idea of how to achieve this, I found the `Building a Coordinate Reference System` topic in the tutorials related to this, but I didn't find anything for adding affine operation.

**Comment by snowman2** [comment]
What does your custom CRS look like if you to `crs.to_proj_json()`?

**Reply by originlake** [reply]
Thanks for your response. I don't have a CRS object because I don't know how to properly constructed one with the affine operation. I directly create a pipeline to do the transformation for now. The local calibration is usually applied on top of a known CRS. For example, EPSG:32616 + affine transform, I can create a pipeline like 
```
+proj=pipeline +step +proj=utm +zone=16 +datum=WGS84 +units=m +no_defs +type=crs +step +proj=affine +xoff=-465.71555958727384 +s11=0.9716346132177371 +s12=-0.23694422339095386 +yoff=-603.671569941072 +s21=0.23694422339095386 +s22=0.9716346132177371 +zoff=18.708113848150706 +s31=-0.0000182280787 +s32=-0.000007637312
```
But this might be not accurate when transforming it to other crs. I want to know if there is a more appropriate way to do it like creating a derived crs via CRS constructor.

**Reply by snowman2** [reply]
You may get a better answer here: https://lists.osgeo.org/mailman/listinfo/proj/
